uberfire-extensions/uberfire-metadata/uberfire-metadata-backends/uberfire-metadata-backend-elasticsearch/src/main/java/org/uberfire/ext/metadata/backend/elastic/index/ElasticSearchIndexProvider.java

```diff
@@ -344,14 +352,14 @@ public void putMapping(String index,
         checkNotNull("metaObject",
                      metaObject);
         try {
-            this.getClient().admin().indices().prepareCreate(index.toLowerCase()).get();
+            this.getClient().admin().indices().prepareCreate(prepareIndex(index)).get();
         } catch (ResourceAlreadyExistsException ex) {
             logger.debug("Resource Already exists: " + ex.getMessage());
         }
         Map<String, Object> properties = this.createMappingMap(metaObject.getProperties());
         this.getClient().admin().indices()
-                .preparePutMapping(index.toLowerCase())
-                .setType(type.toLowerCase())
+                .preparePutMapping(prepareIndex(index))
+                .setType(prepareIndex(type))
```

Comment:
Just being curious: how did you find out that parameters for `setType` also need to be sanitized? Is it really needed?

Reply:
@jhrcek I guess that types are not going to have this problem, but in the other hand, I can't ensure this is not going to happen, that is why I decide to sanitize them too